lldb/source/Plugins/Language/Swift/SwiftLanguage.cpp

```diff
@@ -856,6 +856,109 @@ SwiftLanguage::GetHardcodedSynthetics() {
           }
           return nullptr;
         });
+    g_formatters.push_back([](lldb_private::ValueObject &valobj,
+                              lldb::DynamicValueType dyn_type,
+                              FormatManager &format_manager)
+                               -> lldb::SyntheticChildrenSP {
+      // Try to recognize a Swift type wrapped in a C++ interop wrapper class.
+      // These types have a typedef from a char to the swift mangled name, and a
+      // static constexpr char field whose type is the typedef, and whose name
+      // is $__swift_mangled_name.
+
+
+      // This only makes sense for Clang types.
+      auto type = valobj.GetCompilerType();
+      auto tsc = type.GetTypeSystem().dyn_cast_or_null<TypeSystemClang>();
+      if (!tsc)
+        return nullptr;
+
+      ProcessSP process_sp(valobj.GetProcessSP());
+      auto *swift_runtime = SwiftLanguageRuntime::Get(process_sp);
+      if (!swift_runtime)
+        return nullptr;
+
+      auto module = valobj.GetModule();
+      if (!module) {
+        module = valobj.GetTargetSP()
+                     ->GetExecutableModule();
+        if (!module)
+          return nullptr;
+      }
+
+      auto gts = module->GetTypeSystemForLanguage(
+          lldb::LanguageType::eLanguageTypeSwift);
+      if (!gts)
+        return nullptr;
+
+      auto *ts = llvm::cast<TypeSystemSwift>(gts->get());
+
+      // We operate directly on the qualifier type because the TypeSystem
+      // interface doesn't allow us to check for static constexpr members.
+      auto qual_type = TypeSystemClang::GetQualType(type.GetOpaqueQualType());
+      auto *record_type = llvm::dyn_cast_or_null<clang::RecordType>(
+          qual_type.getTypePtrOrNull());
+      if (!record_type)
+        return nullptr;
+
+      const clang::RecordDecl *record_decl = record_type->getDecl();
+      CompilerType swift_type;
+
+      for (auto *child_decl : record_decl->decls()) {
+        auto *var_decl = llvm::dyn_cast<clang::VarDecl>(child_decl);
+        if (!var_decl)
+          continue;
+
+        auto name = var_decl->getName();
+        if (name != "$__swift_mangled_name")
+          continue;
+
+        const auto *typedef_type =
+            llvm::dyn_cast<clang::TypedefType>(var_decl->getType());
+        if (!typedef_type)
+          continue;
+
+        auto *decl = typedef_type->getDecl();
+        if (!decl)
+          continue;
+
+        auto swift_name = decl->getName();
+        if (!swift::Demangle::isMangledName(swift_name))
+          continue;
+
+        swift_type = ts->GetTypeFromMangledTypename(ConstString(swift_name));
```

Comment:
Same question as above, since all that needs to go in the using C++ dylib are the accessors for the wrapped Swift object, do you know that they will necessarily inject the type into this module?  Otherwise, this should be a global search (though it makes sense to start with the value object's module...)

Reply:
That's a good point... Do we handle a similar case somewhere else that I could take a look at? Otherwise how would you implement this?

Reply:
Maybe value object's module, then executable, then the first module we can find that handles swift..?

Reply:
You certainly want to start in the value object's module, we always try to prioritize the "most proximate" match whenever we do lookups.  Plus it's a likely place to look, and so is a fast path.

After that it depends on the semantics of the language.  For something like C++ where we pretend the ODR rule is taken seriously, we find the first match in all modules and stop at that.  I think Swift has to work that way as well - you can ask the runtime to give you a type for a mangled name, which wouldn't work if there were two viable types for the same mangled name.  So you just start looking at the beginning of the module list and keep going till you find a module that matches that mangled name.

I'm not quite sure why you need a module, for the most part we get types from the Swift Runtime, don't we?  So the module doesn't seem essential?  But I'm not 100% up to date on how type searches are done in swift these days...